src/plugins/data/common/search/aggs/agg_config.test.ts

```diff
@@ -677,9 +687,41 @@ describe('AggConfig', () => {
         },
       };
       const aggConfig = ac.createAggConfig(configStates);
-      const json = aggConfig.toExpressionAst()?.arguments.json;
+      const json = aggConfig.toExpressionAst()?.chain[0].arguments.json;
       expect(json).toEqual([JSON.stringify(configStates.params.json)]);
     });
+
+    it('stringifies arrays only if they are objects', () => {
+      const ac = new AggConfigs(indexPattern, [], { typesRegistry });
+      const configStates = {
+        type: 'range',
+        params: {
+          field: 'bytes',
+          ranges: [
+            { from: 0, to: 1000 },
+            { from: 1001, to: 2000 },
+            { from: 2001, to: 3000 },
+          ],
+        },
+      };
+      const aggConfig = ac.createAggConfig(configStates);
+      const ranges = aggConfig.toExpressionAst()?.chain[0].arguments.ranges;
+      expect(ranges).toEqual([JSON.stringify(configStates.params.ranges)]);
+    });
+
+    it('does not stringify arrays which are not objects', () => {
+      const ac = new AggConfigs(indexPattern, [], { typesRegistry });
+      const configStates = {
+        type: 'percentiles',
+        params: {
+          field: 'bytes',
+          percents: [1, 25, 50, 75, 99],
+        },
+      };
+      const aggConfig = ac.createAggConfig(configStates);
+      const percents = aggConfig.toExpressionAst()?.chain[0].arguments.percents;
+      expect(percents).toEqual([1, 25, 50, 75, 99]);
+    });
```

Comment:
Uncovered a bug in `toExpressionAst` where the `percents` arg was incorrectly stringified when we should have preserved it, so I updated the logic to handle this case where a param is already an array, but not an object array.